src/Protocol/Protocol.h

```diff
@@ -63,6 +64,12 @@ class cProtocol
 	/** Called when client sends some data */
 	virtual void DataReceived(const char * a_Data, size_t a_Size) = 0;
 
+	/** Called when client sends some data over the FML|HS plugin channel */
+	void ForgeDataReceived(const char * a_Data, size_t a_Size)
+	{
+		m_Client->m_ForgeHandshake.DataReceived(m_Client, a_Data, a_Size);
+	}
```

Comment:
Since this is only ever called from within the `cClientHandle`, why does it have to go through `cProtocol`? It brings in many dependencies into this header file needlessly.